app/krs.js

```diff
@@ -78,38 +110,52 @@ exports.provisionKey = function(req) {
         process.config.requesterAuth.clients[req.body.requesterId] !== req.body.requesterSecret) {
       throw utils.ErrorResponse(401, 'invalid requesterSecret');
     }
-    key.requesterId = req.body.requesterId;
   }
 
-  return key.saveQ()
-  .then(function() {
-    if (req.body.disableKRSEmail) {
-      // for API users who don't want their inboxes filling up with unread KRS backup key creation emails
-      return;
+  let masterKey = yield MasterKey.findOne({ customerId, coin });
+
+  if (!masterKey) {
+    masterKey = yield provisionMasterKey(coin, customerId);
+  }
+
+  key.masterKey = masterKey.xpub;
+  key.path = `m/${masterKey.keyCount}`;
+  key.xpub = utils.deriveChildKey(key.masterKey, key.path);
+
+  key.userEmail = req.body.userEmail;
+  key.notificationURL = req.body.notificationURL;
+
+  key.custom = req.body.custom || {};
```

Comment:
I'm confused -- in www, the old krs code called it "userCustom" -- I'd like to keep the same naming convention to minimize the amount of changed code for sending basically the same data. Confusion comes from this file looking like it used to call it just "custom" too -- maybe www is wrong and it's just always come through as null lol

Reply:
worked this out with david off band